examples/sdk-upload/README.md

```diff
@@ -0,0 +1,334 @@
+  assert(!err) // not much point continuing without plugins ...
+
+  // adding our routes (we'll be creating this file next)
+  server.route(require('./routes.js'))
+
+  // starting the server
+  server.start(function (err) {
+    assert(!err) // not much point continuing if the server does not start ...
+    console.log('The server is running on: ', server.info.uri)
+  })
+})
+
+module.exports = server
+```
+
+Next create a file called `routes.js` and add the following:
+
+```js
+// require the function we just created
+var s3 = require('../src/upload.js')
+// path is needed to resolve the filepath to our index.html
+var path = require('path')
+
+module.exports = [
+  // our index route
+  {
+    method: 'GET',
+    path: '/',
+    handler: function (request, reply) {
+      return reply.file(path.resolve(__dirname, '../public/index.html'))
+    }
+  },
+  // our endpoint
+  {
+    method: 'POST',
+    path: '/file_submitted',
+    handler: function (request, reply) {
+      // we'll be sending through a file along with it's filename
+      var file = request.payload.file
+      var filename = request.payload.filename
+      // upload the file to S3
+      s3.upload(file, filename, function (err, data) {
+        if (err) console.log(err)
+        // send the data back to the front end
+        reply(data)
+      })
+    }
+  },
+  // this is for our static files
+  {
+    method: 'GET',
+    path: '/{param*}',
+    handler: {
+      directory: {
+        path: path.resolve(__dirname, '../public'),
+        listing: true,
+        index: false
+      }
+    }
+  }
+]
+```
+
+### Step 6 - Write the client side code to send our file to the server
+
+Create a new directory called `public`. Create an `index.html` file within this
+directory. Add the following to the file:
+
+```html
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>S3 Upload Demo</title>
+    <link rel="icon" href="http://downloadicons.net/sites/default/files/upload-icon-46097.png">
+    <!-- optional stylesheet -->
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/flat-ui/2.2.2/css/flat-ui.min.css">
+    <!-- link to our client side js file that we will create next -->
+    <script type="text/javascript" src="client.js"></script>
+  </head>
+  <body>
+    <h1>S3 Direct Upload Demo</h1>
+    <br/>
+    <!-- simple form with a file input -->
+    <form>
+      <input class="form-control input-sm" id="fileInput" type="file" name="file" onchange="uploadDemo.saveFile(this.files)"/>
+    </form>
+    <br/>
+    <!-- submit button that will trigger our upload -->
+    <button id="submit" class="btn btn-embossed btn-primary" onclick="uploadDemo.submitFile()">Submit</button>
+    <!-- this will hold the link to our newly uploaded image -->
+    <div class="successMessageContainer">
+      <a class="imageLink"></a>
+    </div>
+  </body>
+</html>
+```
+
+Create the javascript file that we're including in our html file and call it
```

Comment:
@jackcarlisle this _works_ but requires JS... we can do better (_progressive enhancement_) ... 😉 
It's totally my fault for not being _specific_ in the requirements. 
Please create an issue in the backlog to add a `progressive enhancement` version.